GPU/Software/Rasterizer.cpp

```diff
@@ -1451,26 +1451,35 @@ void ClearRectangle(const VertexData &v0, const VertexData &v1)
 	minY = std::max(minY, (int)TransformUnit::DrawingToScreen(scissorTL).y);
 	maxY = std::max(0, std::min(maxY, (int)TransformUnit::DrawingToScreen(scissorBR).y + 16));
 
+	constexpr int MIN_LINES_PER_THREAD = 32 * 16;
+
 	const int w = (maxX - minX) / 16;
 	if (w <= 0)
 		return;
 
 	if (gstate.isClearModeDepthMask()) {
-		ScreenCoords pprime(minX, minY, 0);
 		const u16 z = v1.screenpos.z;
 		const int stride = gstate.DepthBufStride();
 
-		for (pprime.y = minY; pprime.y < maxY; pprime.y += 16) {
-			DrawingCoords p = TransformUnit::ScreenToDrawing(pprime);
-
-			if ((z & 0xFF) == (z >> 8)) {
-				u16 *row = &depthbuf.as16[p.x + p.y * stride];
-				memset(row, z, w * 2);
-			} else {
-				for (int x = 0; x < w; ++x) {
-					SetPixelDepth(p.x + x, p.y, z);
+		if ((z & 0xFF) == (z >> 8)) {
+			ParallelRangeLoop(&g_threadManager, [=](int y1, int y2) {
```

Comment:
I find it a little risky to use coordinates multiplied by 16 as range inputs to ParallelRangeloop, it might get the idea to divide by some odd number of CPUs and you might end up some kind of fractional lines here? Not sure if an actual issue though..

Reply:
Since the min thread size is a multiple of 16, this can't happen.

-[Unknown]

Reply:
I don't see how it can't happen. Let's say it's a 48 pixel high rectangle and it gets split into two tasks, with 24 entries each? At least we end up relying on details of the splitting algorithm..

Reply:
Okay, I divided it out so there's no concern of details.

-[Unknown]

Reply:
thanks! I think the code is slightly clearer now too.